scripts/travisci/check_flake8.sh

```diff
@@ -89,21 +83,37 @@ E731 # do not assign a lambda expression, use a def
 ignored_errors_incremental=(
 )
 
+# DOCSTRING
+# The following error code enables all the error codes for docstring defined
+# here:
+# http://pep257.readthedocs.io/en/latest/error_codes.html
+# If one of the errors needs to be ignored, just add it to the ignore array.
+errors_incremental_doc="${errors_incremental[@]} D"
+
 errors_incremental="${errors_incremental[@]}"
 ignored_errors_incremental="${ignored_errors_incremental[@]}"
 if [[ "${TRAVIS_PULL_REQUEST}" != "false" && "${TRAVIS}" == "true" ]]; then
-  git diff "${TRAVIS_COMMIT_RANGE}" \
+  git diff "${TRAVIS_COMMIT_RANGE}" --diff-filter=M \
```

Comment:
This script would be much easier to read if you split it into 2 stages

1) Build lists of files changed or added, e.g. `files_changed` `files_added`
1) Apply different checks to those files

As you have implemented it, you're actually only running flake8 over *lines changed* in both of those cases (that's what `flake8 diff` does).

Reply:
Done